torch/fx/graph_module.py

```diff
@@ -185,8 +187,21 @@ def recompile(self) -> None:
         """
         self.code = self._graph.python_code(root_module='self')
         cls = type(self)
+
         cls.forward = _forward_from_src(self.code)
 
+        cls_call = cls.__call__
+
+        def print_full_traceback(exctype, value, tb):
+            traceback.print_exception(exctype, value, tb)
+
+        def wrapped_call(self, *args, **kwargs):
+            sys.excepthook = print_full_traceback
+            out = cls_call(self, *args, **kwargs)
+            sys.excepthook = sys.__excepthook__
```

Comment:
How about:

```
            old_excepthook = sys.excepthook
            try:
                sys.excepthook = print_full_traceback
                return cls_call(self, *args, **kwargs)
            finally:
                sys.excepthook = old_excepthook
```

To 1) preserve another excepthook if another one was installed and 2) Make sure we reset things if the forward() call throws